spec/puppetserver/ca/action/list_spec.rb

```diff
@@ -19,6 +19,20 @@
                   "fingerprint"=>"onetwo", "fingerprints"=>{"SHA1"=>"one", "SHA256"=>"onetwo", "SHA512"=>"three", "default"=>"four"}}]}
 
   describe 'error handling' do
+    it 'exit-1 when run on a non-CA, all' do
+      allow(action).to receive(:get_certs_or_csrs).and_return(nil)
+      exit_code = action.run({'all' => true})
+      expect(exit_code).to eq(1)
+      expect(err.string).to include('Error while getting certificates')
+    end
+
+    it 'exit-1 when run on a non-CA, certnames' do
+      allow(action).to receive(:get_certs_or_csrs).and_return(nil)
+      exit_code = action.run({'certname' => ['foo','baz']})
+      expect(exit_code).to eq(1)
+      expect(err.string).to include('Error while getting certificates')
+    end
+
```

Comment:
Could you add a case for getting just the certificate requests?

Reply:
Submitted.  I am not a ruby person so I only covered what mine introduced.  You had+have an existing coverage gap on that piece of the function, but I'm not qualified to be the fixer.